e2e/cypress/integration/api/iam/04_teams_api.spec.ts

```diff
@@ -0,0 +1,324 @@
+  let adminToken = '';
+  const defaultAdminReq = {
+    auth: { bearer: adminToken },
+    method: 'GET',
+    url: '/apis/iam/v2beta/teams'
+  };
+  const cypressPrefix = 'test-teams-api';
+  const now = Cypress.moment().format('MMDDYYhhmm');
+  const project1 = {
+      id: `${cypressPrefix}-project1-${now}`,
+      name: 'Test Project 1'
+  };
+  const project2 = {
+      id: `${cypressPrefix}-project2-${now}`,
+      name: 'Test Project 2'
+  };
+
+  before(() => {
+    cy.adminLogin('/').then(() => {
+      const admin = JSON.parse(<string>localStorage.getItem('chef-automate-user'));
+      adminToken = admin.id_token;
+      defaultAdminReq.auth.bearer = adminToken;
+      cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix,
+        ['policies', 'projects', 'teams']);
+
+      for (const project of [project1, project2]) {
+        cy.request({
+            auth: { bearer: adminToken },
+            method: 'POST',
+            url: '/apis/iam/v2beta/projects',
+            body: project
+        });
+      }
+    });
+  });
+
+  after(() => {
+    cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix,
+      ['policies', 'projects', 'teams']);
+  });
+
+  describeIfIAMV2p1('project assignment enforcement', () => {
+    let nonAdminToken = '';
+    const nonAdminTokenID = `${cypressPrefix}-nonadmin-token-${now}`;
+    const teamID = `${cypressPrefix}-team-${now}`;
+    const defaultNonAdminReq = {
+      headers: { 'api-token': nonAdminToken },
+      method: 'POST',
+      url: '/apis/iam/v2beta/teams'
+    };
+
+    before(() => {
+      cy.request({ ...defaultAdminReq,
+        method: 'POST',
+        url: '/apis/iam/v2beta/tokens',
+        body: {
+            id: nonAdminTokenID,
+            name: 'Nonadmin Token'
+        }
+      }).then((response) => {
+          nonAdminToken = response.body.token.value;
+          defaultNonAdminReq.headers = { 'api-token': nonAdminToken };
+      });
+
+      cy.request({ ...defaultAdminReq,
+        url: '/apis/iam/v2beta/policies',
+        method: 'POST',
+        body: {
+            name: 'token-access',
+            id: `${cypressPrefix}-token-access-policy-${now}`,
+            members: [ `token:${nonAdminTokenID}` ],
+            statements: [
+              {
+                effect: 'ALLOW',
+                actions: [ 'iam:teams:*' ],
+                projects: [ '*' ]
+              },
+              {
+                effect: 'ALLOW',
+                actions: [ 'iam:projects:assign' ],
+                projects: [ project1.id ]
+              }
+            ],
+            projects: []
+        }
+      });
+    });
+
+    beforeEach(() => {
+      cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix, ['teams']);
+    });
+
+    afterEach(() => {
+      cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix, ['teams']);
+    });
+
+    describe('POST /apis/iam/v2beta/teams', () => {
+      context('both superadmin and project1 admin can', () => {
```

Comment:
For all the test names where you mention "project1", keep in mind that someone just looking at the test results will only see test names where "project1" will not be meaningful, so would be slightly better to just talk about "project" generically.

Reply:
I thought about that but it got super wordy in practice. We've already done a lot of iteration here so let's circle back if we think things are still unclear.